### PR TITLE
Validated item type with specs module

### LINKED ISSUE #690: Report invalid item-types with more human-readable list
With `planet data search` if I do an improper item-type I get a response like:

```
Error: Invalid value for 'ITEM_TYPES': ['psscend'] should be one of {'PSScene', 'MOD09GA', 'MYD09GA', 'MYD09GQ', 'SkySatCollect', 'REOrthoTile', 'REScene', 'MOD09GQ', 'PSScene4Band', 'Sentinel1', 'Sentinel2L1C', 'SkySatScene', 'Landsat8L1G', 'PSOrthoTile', 'PSScene3Band'}
```

The { and ' characters make it feel like a dump of a program. It'd be better to have it look like a more natural list. 

`planet orders request` does it in a more readable way, so likely can just copy theirs: 
```
Error: Invalid value: item_type - 'skysatcollectd' is not one of 'PSScene', 'Landsat8L1G', 'SkySatScene', 'SkySatCollect', 'PSOrthoTile', 'REOrthoTile', 'PSScene3Band', 'Sentinel2L1C'.
```

### PR DESCRIPTION
**Pull request by kevinlacaille**
Addresses @jreiberkyle's two comments ([comment 1](https://github.com/planetlabs/planet-client-python/pull/695/files/4935b11afab9a5c731831f6930d5bcc2ab5783fd#r977967575), [comment 2](https://github.com/planetlabs/planet-client-python/pull/695/files/4935b11afab9a5c731831f6930d5bcc2ab5783fd#r977970613)).

Fixed a linting issue pulled in from main. 

**Previous behaviour:**
```
Error: Invalid value for 'ITEM_TYPES': ['boo'] should be one of {'PSScene', 'MOD09GA', 'MYD09GA', 'MYD09GQ', 'SkySatCollect', 'REOrthoTile', 'REScene', 'MOD09GQ', 'PSScene4Band', 'Sentinel1', 'Sentinel2L1C', 'SkySatScene', 'Landsat8L1G', 'PSOrthoTile', 'PSScene3Band'}
```

**New behaviour:**
```
Usage: planet data search [OPTIONS] ITEM_TYPES [FILTER]
Try 'planet data search --help' for help.

Error: Invalid value for 'ITEM_TYPES': item_type - 'boo' is not one of 'Landsat8L1G', 'SkySatScene', 'REOrthoTile', 'PSScene', 'PSOrthoTile', 'REScene', 'SkySatCollect', 'PSScene4Band', 'Sentinel1', 'MOD09GQ', 'MOD09GA', 'MYD09GA', 'MYD09GQ', 'PSScene3Band', 'Sentinel2L1C'.
```

Closes #690

### CODE REVIEW

**Comment by jreiberkyle** [comment]
@kevinlacaille I'm a bit concerned that a linting issue was able to pass CI checks and make it into main. I'm going to spend a little bit of time seeing if I can track down the cause. I can confirm that `main` branch as it is does not pass a local run of `nox -s lint`

**Comment by jreiberkyle** [comment]
for now, I'm going to just chalk that issue up to the IP PRs having their checks run prior to the flake8 lint change. Will keep this in mind but I feel good that the CI of this PR picked up the linting issue.

File: planet/cli/data.py [hunk 1]
```diff
@@ -65,15 +65,9 @@ def assets_to_filter(ctx, param, assets: List[str]) -> Optional[dict]:
...
-        [a.lower() for a in ALL_ITEM_TYPES])
-    if invalid_item_types:
-        raise SpecificationException(invalid_item_types,
-                                     ALL_ITEM_TYPES,
-                                     'item_type')
-    else:
-        return item_types
+    for item_type in item_types:
+        validate_item_type(item_type)
+    return item_types
```
[the remark this answers is not in the document]
Comment (jreiberkyle) [reply planet/cli/data.py:70]: An added benefit of adding the output to the PR desc: easy to confirm or deny that the actual output is what is desired in #690. Currently, it's not yet a match.

**Review by jreiberkyle — CHANGES_REQUESTED** [review]
The CLI output does not match what is requested in #690, this can likely be fixed by raising `click.BadParameter` instead of `SpecificationException`

File: planet/cli/data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jreiberkyle) [reply planet/cli/data.py:70]: here's an example of catching/throwing: https://github.com/planetlabs/planet-client-python/blob/3aa079918b098e9ff6bae346f64d8bd87db5a9d7/planet/cli/orders.py#L301

File: tests/integration/test_orders_cli.py [hunk 2]
```diff
@@ -514,8 +516,9 @@ def test_cli_orders_request_product_bundle_incompatible(invoke):
         '--name=test',
         '--id=4500474_2133707_2021-05-20_2419',
     ])
+    error_msg = "Usage: main orders request [OPTIONS] ITEM_TYPE BUNDLE"
```
[the remark this answers is not in the document]
Comment (kevinlacaille) [reply tests/integration/test_orders_cli.py:519]: You got it!

File: planet/cli/data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinlacaille) [reply planet/cli/data.py:70]: Ah yes, I didn't catch this - thank you for the example!

File: tests/integration/test_orders_cli.py [hunk 3]
```diff
@@ -502,8 +503,9 @@ def test_cli_orders_request_product_bundle_invalid(invoke):
...
```
[the remark this answers is not in the document]
Comment (kevinlacaille) [reply tests/integration/test_orders_cli.py:506]: I completely agree, but this isn't an addition to the Data CLI PR, it's just a linting error I've resolved in this PR. If we want to change the Orders CLI's error message, I'd say let's create a new ticket for it :)

File: planet/cli/data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinlacaille) [reply planet/cli/data.py]: Catching `specs.SpesificationError` and raising `click.BadParameter` does indeed solve the problem - thanks @jreiberkyle !

**Review by jreiberkyle — APPROVED** [review]
looks great! And thank you so much for adding the behavior change to the PR desc. super helpful!

**Pull request merged**